fanctrl.py

```diff
@@ -252,51 +290,122 @@ def run(self, debug=True):
-    )
-
-    runGroup = parser.add_argument_group("run")
-    runGroup.add_argument("--run", help="run the service", action="store_true")
-    runGroup.add_argument("--config", type=str, help="Path to config file", default=DEFAULT_CONFIGURATION_FILE_PATH)
-    runGroup.add_argument(
-        "--no-log", help="Disable print speed/meanTemp to stdout", action="store_true"
-    )
-    commandGroup = parser.add_argument_group("configure")
-    commandGroup.add_argument(
-        "--query", "-q", help="Query the currently active strategy", action="store_true"
-    )
-    commandGroup.add_argument(
-        "--list-strategies", help="List the available strategies", action="store_true"
-    )
-    commandGroup.add_argument(
-        "--reload", "-r", help="Reload the configuration from file", action="store_true"
-    )
-    commandGroup.add_argument("--pause", help="Pause the program", action="store_true")
-    commandGroup.add_argument("--resume", help="Resume the program", action="store_true")
+        # safety fallback to avoid damaging hardware
+        if len(temps) == 0:
+            return 50
+
+        return round(temps[0], 1)
 
+
+class FanControllerKmod(FanController):
+    get_fanspeed_files: List[Path]
+    set_fanspeed_files: List[Path]
+    reset_to_auto_files: List[Path]
+    get_temp_files: List[Path]
+
+    def __init__(self, config: Configuration, strategyName: str):
+        super().__init__(config)
+
+        self.sysfs_get_fanspeed_files = []
+        self.sysfs_set_fanspeed_files = []
+        self.sysfs_reset_to_auto_files = []
+        self.sysfs_get_temp = []
+
+        fan_ctrl_base_dir = Path('/sys/class/hwmon')
+        if not fan_ctrl_base_dir.exists() or not fan_ctrl_base_dir.is_dir():
+            raise KModNotWorkingException(f"{str(fan_ctrl_base_dir)} does not exist or is not a directory")
+
+        for hwmon_interface in fan_ctrl_base_dir.iterdir():
+            name_file = hwmon_interface / "name"
+            with name_file.open("r") as name_fp:
+                device_name = name_fp.read().strip()
+            if device_name == "framework_laptop":
+                for file in hwmon_interface.iterdir():
+                    if file.name.startswith("fan") and file.name.endswith("_target"):
+                        self.sysfs_get_fanspeed_files.append(file)
+                    if file.name.startswith("pwm"):
+                        if file.name[3:].isdecimal():
+                            self.sysfs_set_fanspeed_files.append(file)
+                        elif file.name.endswith("_enable"):
+                            self.sysfs_reset_to_auto_files.append(file)
+            elif device_name == "acpitz":
+                for file in hwmon_interface.iterdir():
+                    if file.name.startswith("temp") and file.name.endswith("_input"):
+                        self.sysfs_get_temp.append(file)
+            elif self.configuration.getUseGPU() and device_name == "amdgpu":
```

Comment:
> it didn't include the dGPU temperature before
In fact, it do since March
You can disable it if you want, but it definitely should be on by default.
If this option stays, could you please also document it in the README, and add its default state in the config.json?